xwrf/diagnostics.py

```diff
@@ -0,0 +1,67 @@
+"""Provide lazy calculation of physical variables from esoteric WRF components.
+
+Only four fields are implemented here:
+- air_potential_temperature (from T)
+- air_pressure (from P and PB)
+- geopotential (from PH and PHB)
+- geopotential_height (from geopotential)
+
+All other typical diagnostics (like air_temperature, etc.) are calculable using these fields (and
+those otherwise present) using a general-purpose toolkit.
+"""
+
+
+def _check_if_dask(dataarray):
+    # TODO: this needs to actually do something, however, TBD how to work through the chunks arg
+    # in the backend API (otherwise, weird conflicts may occur)
+    return True
+
+
+def calc_base_diagnostics(dataset, drop=True):
+    """Calculate the four basic fields that WRF does not have in physically meaningful form.
+
+    Parameters
+    ----------
+    dataset : xarray.Dataset
+        Dataset representing WRF data opened via NetCDF4 backend with chunking.
+    drop : bool
+        Decide whether to drop the components of origin after creating the diagnostic fields from
+        them.
+
+    Notes
+    -----
+    This operation should be called before destaggering or any other cleaning operation.
+    """
+    # Potential temperature
+    if _check_if_dask(dataset['T']):
```

Comment:
Should this only work if `dask` is present?

I guess we should either
- only allow `xwrf` usage with `dask` -> check somewhere earlier for dask (like in `__init__`)
- also allow `xwrf` usage without `dask` -> remove these checks

Reply:
This is really just a placeholder to capture the intent of needing a Dask check of some kind before diagnostics are computed. It would be a good conversation to have as to what level of Dask-based operations are required vs. optional, and if optional, how those choices are specified.